lib/livebook_web/live/session_live.ex

```diff
@@ -1188,6 +1191,21 @@ defmodule LivebookWeb.SessionLive do
     {:noreply, socket}
   end
 
+  def handle_info({:env_vars_changed, env_vars}, socket) do
+    for env_var <- socket.assigns.env_vars do
+      System.delete_env(env_var.key)
+    end
+
+    socket =
+      socket
+      |> assign(env_vars: env_vars)
+      |> put_env_vars()
+
+    maybe_reconnect_runtime(socket)
```

Comment:
I'm triggering the runtime reconnection so users doesn't need to evaluate the setup again. But, is it a good idea?

Reply:
We definitely don't want to restart the runtime, because this way they may loose some expensive computations. We should set the secrets in the runtime directly with `Runtime.put_system_envs` (this way they also don't leak into the Livebook node unnecessarily :)).

Reply:
So, it's almost the same as the secrets does, right?

Reply:
It's similar, but we don't need to store envs in `Session.Data`, we can manage all of it in `Session`, because it's user-agnostic.